### PR TITLE
Update versioning section in api design guidelines

### LINKED ISSUE #206: Update of API versioning section in API guidelines
**Problem description**
This update is to align the versioning section with the API versioning process described in release management Wiki.

**Expected action**
Review and agree on the proposed changes for merge into API-design-guidelines.md

**Additional context**
RM Wiki - [API versioning](https://wiki.camaraproject.org/x/a4BaAQ)

### PR DESCRIPTION
**Pull request by tanjadegroot**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:
It updates the versioning section to be in line with the API versioning for the release management process

#### Which issue(s) this PR fixes:

Fixes #206 

#### Changelog input

```
 release-note: update the API versioning in line with the release management process 

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -585,30 +627,36 @@ Bearing in mind that APIs are continually evolving and certain operations will n
 - Remove deprecated APIs documentation.
 - Never start using already deprecated APIs.
 
+<font size="3"><span style="color: blue"> Types of modification: </span></font>
 
-<font size="3"><span style="color: blue"> Types of modification </span></font>
+- Not all API changes have an impact on API consumers. Such changes are referred to as backward compatible changes.
+- If the API undergoes changes of this type, it shall be provided through a maintenance-release which will replace the current one.
```
Comment (hdamker) [inline documentation/API-design-guidelines.md:633]: I don't understand this line. My understanding is that backward compatible changes can happen:
* as part of patch release, e.g. to correct a bug within a minor version
* between minor releases (e.g. v1.2.0 and v1.3.0), e.g. introduction of new functionality without a breaking change

I suppose that the second case is not covered by "maintenance-release"?

**Review by hdamker — COMMENTED** [review]
Looks good to me. One proposal and one question below.

**Review by hdamker — DISMISSED** [review]

**Comment by rartych** [comment]
I have discovered that in Commonalities we have https://github.com/camaraproject/Commonalities/blob/main/documentation/Camara_Versioning_Guidelines.md
I think it should be deprecated with redirection to the proposed here API Design Guidelines and Release Management documentation.
To speed up things maybe we can do it within this PR. @tanjadegroot @hdamker ?

**Comment by tanjadegroot** [comment]
Hi Rafal, yes that is correct. this file can be deprecated.
API versioning is now going into the API Guidelines section 5
API release process is now here: https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/API_Release_Guidelines.md (with more details in the CAMARA release Management Wiki)

**Comment by rartych** [comment]
Tanja, if you could modify the file in Nokia fork: https://github.com/nokia/camaraproject-Commonalities/blob/Update-versioning-section-in-API-design-guidelines/documentation/Camara_Versioning_Guidelines.md then we go with all the changes within this PR.

**Comment by tanjadegroot** [comment]
@hdamker to check this update

**Comment by tanjadegroot** [comment]
> Tanja, if you could modify the file in Nokia fork: https://github.com/nokia/camaraproject-Commonalities/blob/Update-versioning-section-in-API-design-guidelines/documentation/Camara_Versioning_Guidelines.md then we go with all the changes within this PR.

I just need to remove the file corrrect ?

**Comment by hdamker** [comment]
> I just need to remove the file corrrect ?

@tanjadegroot that's not enough, also the line in the table within https://github.com/camaraproject/Commonalities/blob/main/README.md need to be updated/replaced.

What about deleting the two lines about [API-Readiness-Checklist.md](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md) and [Camara_Versioning_Guidelines.md](https://github.com/camaraproject/Commonalities/blob/main/documentation/Camara_Versioning_Guidelines.md) 

and put underneath the table:

```
NOTE: Guidelines for Release Management of API versions, e.g. the API-Readiness-Checklist, are now located within [ReleaseManagement](https://github.com/camaraproject/ReleaseManagement). The versioning of API is defined within API-design-guidelines.md](documentation/API-design-guidelines.md#5-versioning). 
```

@rartych WDYT about the README?

**Review by hdamker — DISMISSED** [review]

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -537,78 +539,125 @@ These considerations are below:
...
+The OAS file also defines the API version used in the URL of the API endpoint (in the servers object).
+
+The API version in the URL only includes the "x" (MAJOR version) number of the API version as follows:
+
+```yaml
+servers:
+    url: {apiRoot}/qod/v2
 ```
 
+NOTE: CAMARA exceptionally allows to use both the MAJOR and the MINOR version number (v0.y) separated by a dot (".") in initial API versions (x=0) in the API version in the URL.
```
Comment (shilpa-padgaonkar) [inline documentation/API-design-guidelines.md:584]: The maturity states initial & stable are described only later in the doc L598. Would be better to have this info before it being used here.

IMHO,  the statement "CAMARA exceptionally allows to use both the MAJOR and the MINOR version number" does not clearly state that this is a "requirement" by Camara.

Would prefer a simpler format --- Camara APIs with initial maturity state **shall** use both the MAJOR and the MINOR version number (v0.y) separated by a dot (".") in the API version ......

Comment (tanjadegroot) [reply documentation/API-design-guidelines.md:584]: Was it agreed that this is a MUST ?

Comment (hdamker) [reply documentation/API-design-guidelines.md]: @tanjadegroot we had the discussion, but might have taken no final decision. But I support the clear line to make it mandatory - using v0 without the minor version would only work if there is only one initial version (which is a corner case).

**Review by rartych — DISMISSED** [review]
LGTM

**Review by patrice-conil — APPROVED** [review]
/LGTM

**Review by hdamker — DISMISSED** [review]

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

File: documentation/API-design-guidelines.md [hunk 3]
```diff
@@ -537,78 +539,134 @@ These considerations are below:
...
+- API consumers shall be notified of the new version availability so that they can take it into account.
 
-Not all API changes have an impact on API consumers. These changes are often referred to as backward compatible changes. If the API undergoes changes of this type, it should not be necessary to release a new version, it will suffice to replace the current one. What would be very convenient is to notify our consumers with the new changes so that they take them into account.
+Backward compatible changes to an API that **DO NOT** affect consumers:
 
-This is a list of changes to an API that **DO NOT** affect consumers:
+- Adding a new endpoint
+- Adding new operations on a resource (`PUT`, `POST`, ...).
+- Adding optional input parameters to requests on existing resources. For example, adding a new filter parameter in a GET on a collection of resources.
+- Changing an input parameter from reqmandatory to optional. For example: when creating a resource, a property of said resource that was previously mandatory becomes optional.
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:651]:
```suggestion
- Changing an input parameter from mandatory to optional. For example: when creating a resource, a property of said resource that was previously mandatory becomes optional.
```

**Review by rartych — APPROVED** [review]

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by hdamker — APPROVED** [review]

**Comment by rartych** [comment]
@patrice-conil Could you approve it?

**Review by patrice-conil — APPROVED** [review]
LGTM

**Pull request merged**